packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -48,6 +50,42 @@ function createPrototype<
   return proto;
 }
 
+// preprocess the ontology definition to more quickly apply object conversions when needed
+function createConverter<
+  T extends keyof O["objects"] & string,
+  O extends OntologyDefinition<any>,
+>(
+  ontology: O,
+  type: T,
+) {
+  const steps: Array<(o: Record<string, any>) => void> = [];
+
+  for (
+    const [key, value] of Object.entries(ontology.objects[type].properties)
```

Comment:
There are actually not value types that require recursion (as long as we're OK with doing arrays right when they come in)